megamek/src/megamek/client/bot/princess/FireControl.java

```diff
@@ -2836,4 +2752,39 @@ protected Mounted getAntiAirAmmo(List<Mounted> ammoList, WeaponType weaponType,
 
         return returnAmmo;
     }
+
+    // Helper method that figures out the valid facing changes for the given shooter
+    private List<Integer> getValidFacingChanges(Entity shooter)
+    {
+    	// figure out all valid twists or turret turns
+        // mechs can turn:
+        //		one left, one right unless he has "no torso twist" quirk or is on the ground
+        //		two left, two right if he has "extended torso twist" quirk
+        // vehicles and turrets can turn any direction unless he has no turret
+        ArrayList<Integer> validFacingChanges = new ArrayList<Integer>();
```

Comment:
Should be List rather than ArrayList for the declaration, and actually, this should probably be a Set/HashSet since the order doesn't matter and each facing should be unique.